pkg/cloud/azure/actuators/machine_scope_test.go

```diff
@@ -169,3 +170,69 @@ func TestCredentialsSecretFailures(t *testing.T) {
 		t.Errorf("Expected New credentials secrets to succeed but found : %v", err)
 	}
 }
+
+func TestNewMachineScope(t *testing.T) {
+	machineConfig := providerv1.AzureMachineProviderSpec{
+		Location:          "test",
+		ResourceGroup:     "test",
+		CredentialsSecret: &corev1.SecretReference{Name: "testCredentials", Namespace: "dummyNamespace"},
+	}
+	providerSpecWithValues, err := providerSpecFromMachine(&machineConfig)
+	if err != nil {
+		t.Fatalf("error encoding provider config: %v", err)
+	}
+
+	testCases := []struct {
+		machine               *machinev1.Machine
+		secret                *corev1.Secret
+		expectedLocation      string
+		expectedResourceGroup string
+	}{
+		{
+			machine: &machinev1.Machine{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "test",
+				},
+				Spec: machinev1.MachineSpec{
+					ProviderSpec: *providerSpecWithValues,
+				},
+			},
+			secret: &corev1.Secret{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "testCredentials",
+					Namespace: "dummyNamespace",
+				},
+				Data: map[string][]byte{
+					"azure_subscription_id": []byte("dummySubID"),
+					"azure_client_id":       []byte("dummyClientID"),
+					"azure_client_secret":   []byte("dummyClientSecret"),
+					"azure_tenant_id":       []byte("dummyTenantID"),
+					"azure_resourcegroup":   []byte("dummyResourceGroup"),
+					"azure_region":          []byte("dummyRegion"),
+					"azure_resource_prefix": []byte("dummyClusterName"),
+				},
+			},
+			expectedLocation:      "test",
+			expectedResourceGroup: "test",
+		},
+	}
+
+	for _, tc := range testCases {
```

Comment:
There is only one test case. No need for table driven testing. What does `TestNewMachineScope` mean? Are you testing the location set in machine's provider config overrides the one in the secret? If so, choose different locations in each object so you can validate which has higher priority. Also important to test other cases when the location in the secret is not set. Or, location in the machine provider config is not set.